runtime/lua/vim/diagnostic.lua

```diff
@@ -1589,12 +1562,16 @@ M.handlers.underline = {
 --- @param diagnostics table<integer, vim.Diagnostic[]>
 --- @param opts vim.diagnostic.Opts.VirtualText
 local function render_virtual_text(namespace, bufnr, diagnostics, opts)
+  local lnum = api.nvim_win_get_cursor(0)[1] - 1
   api.nvim_buf_clear_namespace(bufnr, namespace, 0, -1)
 
   for line, line_diagnostics in pairs(diagnostics) do
     local virt_texts = M._get_virt_text_chunks(line_diagnostics, opts)
+    local should_render_current_line = (opts.current_line == true and line == lnum)
```

Comment:
If `opts.current_line` is `nil` this should also be true.

Reply:
That case is handled by the last part of the condition: `or (opts == true or type(opts) == 'table')`

Reply:
`opts` will always be a table here, so `opts == true` will always be false and `type(opts) == 'table'` will always be true.

I think the logic would be simplified here for testing whether or not the line should be skipped:

```lua
local skip = (opts.current_line == true and line ~= lnum) or (opts.current_line == false and line == lnum)
```

Reply:
Ah, thanks, that helps a lot.  I saw examples int he docs of passing in `virtual_text = true` which seemed to be passed directly into `render_virtual_text`, so I thought `true` was a possible value.

```lua
vim.diagnostic.config({ virtual_text = true })
```

```lua
render_virtual_text(
    ns.user_data.virt_text_ns, 
    bufnr, 
    line_diagnostics, 
    opts.virtual_text -- <--- 
)
```